xarray/plot/utils.py

```diff
@@ -41,7 +41,7 @@ def _build_discrete_cmap(cmap, levels, extend, filled):
 
     if not filled:
         # non-filled contour plots
-        extend = 'neither'
+        extend = 'max'
```

Comment:
This is the actual bug fix for https://github.com/pydata/xarray/issues/866 which motivated this PR ;-).

xarray wasn't plotting contours correctly before this change, see the following code:

```python
import numpy as np
import matplotlib.pyplot as plt
import xarray as xr

x, y = np.meshgrid(np.arange(12), np.arange(12))
z = xr.DataArray(np.sqrt(x**2 + y**2))
z.plot.contour(levels=[2, 4, 6, 8], add_colorbar=False, colors='k')
plt.show()
```
![figure_1](https://cloud.githubusercontent.com/assets/10050469/16994818/5779881c-4ea9-11e6-802c-5a0d671517e2.png)

With this fix everything is fine now:

![figure_1-1](https://cloud.githubusercontent.com/assets/10050469/16994804/45776abc-4ea9-11e6-8b19-d1a05a7d1103.png)

Intuitively, it also makes sense: the last contour does mark all data beyond the last level, hence `extend = 'max'`